docs/design/properties_code_point_trie.md

```diff
@@ -57,6 +71,42 @@ A `CodePointTrie` optimizes over a generic inversion map in different ways. One
 
 ## Implementing `CodePointTrie` in ICU4X 
 
+ICU4X benefits by reusing the optimized data structures produced by ICU.
+In practice, this means exporting the code point trie binary data per property from ICU4C.
+As a result, for the Rust enums / new types that represent Unicode properties in ICU4X, the discriminant integer values corresponding to each of the enum variants should match the corresponding enum integer values in ICU4C and ICU4J.
+ICU4X code that can read the ICU code point trie binary data is a port of the original ICU code doing the same.
+The code can traverse the binary (byte array) representation of the trie without any extra heap allocations.
+Code in ICU4X ported from ICU includes getting the value for a code point input (`get()`), and getting the longest range of consecutive code points starting at the input code point whose trie values all match the provided input value (`get_range()`).
+
+### Value Width
+
+ICU `CodePointTrie`s are implemented logically as 2 arrays: an index array, and a data array.
+The index array contains values needed for the iterative 1 or 3 lookups needed to traverse the trie prior to the last lookup.
+Each element of the index array is 16 bits wide.
+
+The penultimate lookup (the last lookup in the index array) returns a value that is the index into the data array.
+The data array stores the values of the trie associated to each code point.
+Data array elements are uniformly wide, with a width that is currently either 8 bits, 16 bits, or 32 bits.
+
+### Trie Type
+
+The features of code point tries in ICU are inherited in ICU4X.
+One is that a trie can either be a "fast" type or a "small" type. 
+The code point trie (ICU 3rd version CodePointTrie) is designed to return a value using either 2 or 4 array lookups.
+For either trie type, there is a limit ("fast max limit") value that partitions the code space into code points that need 2 lookups (`[0x0000, fastMax)`) and code points that need 4 lookups (`[fastMax, 0x10FFFF]`).
+For small type tries, the `fastMax` limit is `0x1000`, and for fast type tries, the `fastMax` limit is `0x10000`.
+In effect, in either trie type, the first 1/16 of the first plane of the Unicode code space—`[0x0000, 0x0FFF]`—will only need 2 lookups.
+Similarly, all code points in the 16 planes remaining planes after the first plane—`[0x10000, 0x10FFFF]`—will always have 4 array lookups.
+Whether a code point trie is a small or fast type only affects the number of array lookups for code points in the range `[0x1000, 0x10000)`.
+[This range](https://unicode.org/roadmaps/bmp/) contains various scripts, symbols, CJK characters, and other East Asian characters.
```

Comment:
Suggestion: Also please improve the docs of `TrieType` to describe this tradeoff better.